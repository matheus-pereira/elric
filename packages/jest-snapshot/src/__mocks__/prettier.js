/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

const prettier = elric.requireActual('prettier');

module.exports = {
  format: (text, opts) =>
    prettier.format(text, {
      pluginSearchDirs: [require('path').dirname(require.resolve('prettier'))],
      ...opts,
    }),
  getFileInfo: {sync: () => ({inferredParser: 'babel'})},
  resolveConfig: {sync: elric.fn()},
  version: prettier.version,
};
