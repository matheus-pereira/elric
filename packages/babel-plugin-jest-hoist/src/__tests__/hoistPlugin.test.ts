/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as path from 'path';
import pluginTester from 'babel-plugin-tester';
import {format as formatCode} from 'prettier';
import babelPluginelricHoist from '..';

pluginTester({
  plugin: babelPluginelricHoist,
  pluginName: 'babel-plugin-elric-hoist',
  tests: {
    'automatic react runtime': {
      babelOptions: {
        babelrc: false,
        configFile: false,
        filename: path.resolve(__dirname, '../file.js'),
        presets: [
          [
            require.resolve('@babel/preset-react'),
            {development: true, runtime: 'automatic'},
          ],
        ],
      },
      code: `
        elric.mock('./App', () => () => <div>Hello world</div>);
      `,
      formatResult(code) {
        // replace the filename with something that will be the same across OSes and machine
        const codeWithoutSystemPath = code.replace(
          /var _jsxFileName = ".*";/,
          'var _jsxFileName = "/root/project/src/file.js";',
        );

        return formatCode(codeWithoutSystemPath, {parser: 'babel'});
      },
      snapshot: true,
    },
    'top level mocking': {
      code: `
        require('x');

        elric.enableAutomock();
        elric.disableAutomock();
      `,
      snapshot: true,
    },
    'within a block': {
      code: `
        beforeEach(() => {
          require('x')
          elric.mock('someNode')
        })
      `,
      snapshot: true,
    },
    'within a block with no siblings': {
      code: `
        beforeEach(() => {
          elric.mock('someNode')
        })
      `,
      snapshot: true,
    },
  },
});
