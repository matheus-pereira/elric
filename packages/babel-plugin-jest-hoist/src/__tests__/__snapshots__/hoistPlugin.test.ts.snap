// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-elric-hoist automatic react runtime: automatic react runtime 1`] = `

elric.mock('./App', () => () => <div>Hello world</div>);

      ↓ ↓ ↓ ↓ ↓ ↓

var _jsxFileName = "/root/project/src/file.js";

_getelricObj().mock(
  "./App",
  () => () =>
    /*#__PURE__*/ _jsxDEV(
      "div",
      {
        children: "Hello world",
      },
      void 0,
      false,
      {
        fileName: _jsxFileName,
        lineNumber: 1,
        columnNumber: 32,
      },
      this
    )
);

import { jsxDEV as _jsxDEV } from "react/jsx-dev-runtime";

function _getelricObj() {
  const { elric } = require("@elric/globals");

  _getelricObj = () => elric;

  return elric;
}


`;

exports[`babel-plugin-elric-hoist top level mocking: top level mocking 1`] = `

require('x');

elric.enableAutomock();
elric.disableAutomock();

      ↓ ↓ ↓ ↓ ↓ ↓

_getelricObj().enableAutomock();

_getelricObj().disableAutomock();

function _getelricObj() {
  const { elric } = require("@elric/globals");

  _getelricObj = () => elric;

  return elric;
}

require("x");


`;

exports[`babel-plugin-elric-hoist within a block with no siblings: within a block with no siblings 1`] = `

beforeEach(() => {
  elric.mock('someNode')
})

      ↓ ↓ ↓ ↓ ↓ ↓

function _getelricObj() {
  const { elric } = require("@elric/globals");

  _getelricObj = () => elric;

  return elric;
}

beforeEach(() => {
  _getelricObj().mock("someNode");
});


`;

exports[`babel-plugin-elric-hoist within a block: within a block 1`] = `

beforeEach(() => {
  require('x')
  elric.mock('someNode')
})

      ↓ ↓ ↓ ↓ ↓ ↓

function _getelricObj() {
  const { elric } = require("@elric/globals");

  _getelricObj = () => elric;

  return elric;
}

beforeEach(() => {
  _getelricObj().mock("someNode");

  require("x");
});


`;
